knowledge-base/adr/0020-generator-options.md

```diff
@@ -37,26 +37,26 @@ We should discuss the details of those defaults and make a concious decision on
+| Current Name, Aliases                                    | Future Name, Aliases             | Current Behavior                                                                                  | Future Behavior                                                                                                                                                                                                                 |
+| :------------------------------------------------------- | :------------------------------- | :------------------------------------------------------------------------------------------------ | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `--inputDir`, `-i` _(required)_                          | `--input`, `-i` _(required)_     | Input directory/file                                                                              | _same_                                                                                                                                                                                                                          |
+| `--outputDir`, `-o` _(required)_                         | `--outputDir`, `-o` _(required)_ | Output directory                                                                                  | _same_                                                                                                                                                                                                                          |
+| `--clearOutputDir`                                       | `--clearOutputDir`               | Delete all files in output directory                                                              | _same_                                                                                                                                                                                                                          |
+| `--generateJs`                                           | `--transpile`, `-t`              | Transpiles, default true                                                                          | Transpiles, default false. If set if tsconfig is enabled with default (unless configured). This should be explicitly stated in the documentation.                                                                               |
+| `--generatePackageJson`                                  | `--packageJson`                  | Writes a default package.json, default true.                                                      | Writes a default package.json, default false. Add a log statment on successful generation, that the core package needs to be installed. Optionally in the future: Writes a custom package.json if passed. Keep boolean for now. |
```

Comment:
I see this as a performance improvement, if the core dependency is the only reason for generating the package json. By the way, the optional custom package.json can also be covered by the new option `--include`.